SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -604,33 +604,41 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 		}
 	}
 
-	// Remove dynamically attached subobjects
+	// Remove dynamic subobjects and objects replicated on the actorchannel.
 	if (USpatialActorChannel* Channel = SpatialNetDriver->GetActorChannelByEntityId(EntityId))
 	{
-		for (UObject* DynamicSubobject : Channel->CreateSubObjects)
+		for (auto& ObjReplicatorPair : Channel->ReplicationMap)
 		{
-			if (FNetworkGUID* SubobjectNetGUID = NetGUIDLookup.Find(DynamicSubobject))
+			TWeakObjectPtr<UObject> WeakObjectPtr = ObjReplicatorPair.Value->GetWeakObjectPtr();
+			if (WeakObjectPtr.IsValid())
```

Comment:
Hm, is it even important for it to be valid? Does the lookup access the underlying object?
I am not sure what the implications would be, but my current intuition is that we are only doing map manipulations, not destroying/checking things.

Reply:
hm, it could be that the validity is important for the lookup if gc affects the internals of the object like with uprops...

Reply:
what's the hash operator for a uobject? Not sure how to find that in code. But I think you are probably right that this isn't needed

Reply:
have removed this now